mmv1/products/integrationconnectors/Connection.yaml

```
@@ -0,0 +1,1120 @@
+self_link: projects/{{project}}/locations/{{location}}/connections/{{name}}
+timeouts: !ruby/object:Api::Timeouts
+  insert_minutes: 30
+  update_minutes: 30
+  delete_minutes: 30
+create_url: projects/{{project}}/locations/{{location}}/connections?connectionId={{name}}
+update_verb: :PATCH
+update_mask: true
+autogen_async: true
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "integration_connectors_connection_basic"
+    primary_resource_id: "pubsubconnection"
+    vars:
+      connection_name: "test-pubsub"
+  - !ruby/object:Provider::Terraform::Examples
+    name: "integration_connectors_connection_advanced"
+    primary_resource_id: "zendeskconnection"
+    vars:
+      connection_name: "test-zendesk"
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: "name"
+    base_url: "{{op_id}}"
+    wait_ms: 1000
+    timeouts: !ruby/object:Api::Timeouts
+      insert_minutes: 50
+      update_minutes: 30
+      delete_minutes: 30
+  result: !ruby/object:Api::OpAsync::Result
+    path: "response"
+    resource_inside_response: true
+  status: !ruby/object:Api::OpAsync::Status
+    path: "done"
+    complete: true
+    allowed:
+      - "true"
+      - "false"
+  error: !ruby/object:Api::OpAsync::Error
+    path: "error"
+    message: "message"
+
+parameters:
+  - !ruby/object:Api::Type::String
+    name: "location"
+    required: true
+    immutable: true
+    url_param_only: true
+    description: |
+      Location in which Connection needs to be created.
+  - !ruby/object:Api::Type::String
+    name: "name"
+    required: true
+    immutable: true
+    url_param_only: true
+    description: |
+      Name of Connection needs to be created.
+
+properties:
+  - !ruby/object:Api::Type::Time
+    name: "createTime"
+    description: |
+      Time the Namespace was created in UTC.
+    output: true
+  - !ruby/object:Api::Type::Time
+    name: "updateTime"
+    description: |
+      Time the Namespace was updated in UTC.
+    output: true
+  - !ruby/object:Api::Type::String
+    name: "description"
+    description: |
+      An arbitrary description for the Conection.
+  - !ruby/object:Api::Type::KeyValueLabels
+    name: "labels"
+    description: |
+      Resource labels to represent user provided metadata.
+  - !ruby/object:Api::Type::String
+    name: "connectorVersion"
+    required: true
+    description: |
+      connectorVersion of the Connector.
+  - !ruby/object:Api::Type::NestedObject
+    name: "status"
+    description: |
+      Status of the Integration Connector.
+    output: true
+    properties:
+      - !ruby/object:Api::Type::Enum
+        name: "state"
+        description: |
+          State of the Integration Connector
+        output: true
+        values:
+          - :STATE_UNSPECIFIED
+          - :CREATING
+          - :ACTIVE
+          - :INACTIVE
+          - :DELETING
+          - :UPDATING
+          - :ERROR
+          - :AUTHORIZATION_REQUIRED
```

```suggestion
      - !ruby/object:Api::Type::String
        name: "state"
        description: |
          State of the Integration Connector
        output: true
```
Since it is an output field and we don't have validation on them. See the generated version downstream: https://github.com/modular-magician/terraform-provider-google-beta/blob/73f1e7f142c6eff78ddb2fc00205355d86265095/google-beta/services/integrationconnectors/resource_integration_connectors_connection.go#L1329

We can also change the rest of enum out put fields to string. It's probably easy to read this way and we also don't need to maintain the values in Terraform, like no need to update here if the API changes possible values for these output fields.

---

Done